src/forms/Form.2.js

```diff
@@ -37,6 +37,8 @@ class Form2 extends Component {
     }
 
     componentDidMount() {
+        let window = this.element.ownerDocument.defaultView;
```

Comment:
... you're passing the DOM default view as a window, to the service? ... I might need some explanation about this.